kratos/processes/apply_periodic_boundary_condition_process.h

```diff
@@ -0,0 +1,196 @@
+#include "geometries/point.h"
+#include "processes/process.h"
+#include "utilities/math_utils.h"
+#include "includes/kratos_parameters.h"
+#include "includes/dof.h"
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+
+class ApplyPeriodicConditionProcess : public Process
+{
+
+  public:
+    /// Pointer definition of ApplyPeriodicConditionProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyPeriodicConditionProcess);
+
+    typedef Dof<double>*                                    DofPointerType;
+    typedef Dof<double>                                     DofType;
+    typedef ModelPart::VariableComponentType                VariableComponentType;
+    typedef KratosComponents<Variable<array_1d<double, 3>>> VectorVariableType;
+    typedef ProcessInfo                                     ProcessInfoType;
+    typedef ProcessInfo::Pointer                            ProcessInfoPointerType;
+    typedef unsigned int                                    IndexType;
+    typedef ModelPart::DoubleVariableType                   VariableType;
+    typedef ModelPart::NodeIterator                         NodeIterator;
+    typedef Element                                         ElementType;
+    typedef Node<3>                                         NodeType;
+    typedef Matrix                                          MatrixType;
+    typedef Vector                                          VectorType;
+    typedef Geometry<NodeType>                              GeometryType;
+
+    /**
+     * @brief Constructor of the process to apply periodic boundary condition
+     * @param rModelPart The main model part where the boundary conditions are applied.
+     * @param rParameters parameters for the periodic condition to be applied
+     */
+    ApplyPeriodicConditionProcess(ModelPart &rMasterModelPart, ModelPart &rSlaveModelPart,
+                                  Parameters rParameters);
+
+    /**
+     * @brief Destructor of the process class
+     */
+    ~ApplyPeriodicConditionProcess();
+
+    /**
+     * @brief Function initializes the process
+     */
+    void ExecuteInitialize() override;
+
+    /**
+     * @brief Function initializes the solution step
+     */
+    void ExecuteInitializeSolutionStep() override;
+
+    /**
+     * @brief Function to print the information about this current process
+     */
+    void PrintInfo(std::ostream& rOStream) const override;
+
+  private:
+    MatrixType mTransformationMatrix; // This can be either for rotating or for translating the slave geometry to Master geometry
+    MatrixType mTransformationMatrixVariable; // This can be either for rotating or for translating the master variable to slave geometry
+    ModelPart &mrMasterModelPart;       // the master modelpart to which the master-slave constraints are added.
+    ModelPart &mrSlaveModelPart;
+    Parameters mParameters;          // parameters
+    double mTheta;
+    bool mIsInitialized;
+    std::vector<double> mCenterOfRotation;
```

Comment:
```suggestion
    DenseVector mCenterOfRotation;
```

Take into account the problematic of the push_backs and other methods exclusive of std::vector

Reply:
yes .. I am not using any push_back after the modifications.